libdd-profiling/src/exporter/profile_exporter.rs

```diff
@@ -0,0 +1,365 @@
+//!
+//! ```no_run
+//! use libdd_profiling::exporter::{config, ProfileExporter};
+//!
+//! # fn main() -> anyhow::Result<()> {
+//! let endpoint = config::file("/tmp/profile_dump.http")?;
+//! let exporter = ProfileExporter::new("dd-trace-test", "1.0.0", "rust", vec![], endpoint)?;
+//! // Requests will be saved to /tmp/profile_dump.http (overwrites on multiple sends)
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! The dumped files contain the complete HTTP request including headers and body,
+//! which can be useful for debugging or replaying requests.
+
+use super::file_exporter::spawn_dump_server_with_builder;
+use anyhow::Context;
+use libdd_common::tag::Tag;
+use libdd_common::{azure_app_services, tag, Endpoint};
+use serde_json::json;
+use std::io::Write;
+use tokio::runtime::Runtime;
+use tokio_util::sync::CancellationToken;
+
+use crate::internal::{EncodedProfile, Profile};
+use crate::profiles::{Compressor, DefaultProfileCodec};
+
+pub struct ProfileExporter {
+    client: reqwest::Client,
+    family: String,
+    base_tags_string: String,
+    request_url: String,
+    headers: reqwest::header::HeaderMap,
+    runtime: Option<Runtime>,
+}
+
+pub struct File<'a> {
+    pub name: &'a str,
+    pub bytes: &'a [u8],
+}
+
+impl ProfileExporter {
+    /// Creates a new exporter to be used to report profiling data.
+    ///
+    /// Note: Reqwest v0.12.23+ includes automatic retry support for transient failures.
+    /// The default configuration automatically retries safe errors and low-level protocol NACKs.
+    /// For custom retry policies, users can configure the reqwest client before creating the
+    /// exporter.
+    ///
+    /// # Thread Safety
+    ///
+    /// The exporter can be used from any thread, but if using `send_blocking()`, the exporter
+    /// should remain on the same thread for all blocking calls. See [`send_blocking`] for details.
+    ///
+    /// [`send_blocking`]: ProfileExporter::send_blocking
+    pub fn new(
+        profiling_library_name: &str,
+        profiling_library_version: &str,
+        family: &str,
+        mut tags: Vec<Tag>,
+        endpoint: Endpoint,
+    ) -> anyhow::Result<Self> {
+        let mut builder = reqwest::Client::builder()
+            .use_rustls_tls()
+            .timeout(std::time::Duration::from_millis(endpoint.timeout_ms));
+
+        let request_url = match endpoint.url.scheme_str() {
+            // HTTP/HTTPS endpoints
+            Some("http") | Some("https") => endpoint.url.to_string(),
+
+            // File dump endpoint (debugging) - uses platform-specific local transport
+            Some("file") => {
+                let output_path = libdd_common::decode_uri_path_in_authority(&endpoint.url)
+                    .context("Failed to decode file path from URI")?;
+                builder = spawn_dump_server_with_builder(builder, output_path)?;
+                "http://localhost/v1/input".to_string()
+            }
```

Comment:
Should we maybe "guard" this behind a feature flag? I like the idea of being able to dump HTTP request header and body to a file, but I am not sure we should have this in release builds. On the other hand, I have no hard feelings against and it seems cool to be able to debug this...
I am unsure.